mlflow/utils/mime_type_utils.py

```diff
@@ -0,0 +1,53 @@
+import os
+import pathlib
+
+from mimetypes import guess_type
+
+# from mlflow.models.model import MLMODEL_FILE_NAME
+# from mlflow.projects._project_spec import MLPROJECT_FILE_NAME
```

Comment:
Importing these constants from here creates a circular import. (artifacts -> utils -> models -> artifacts). Is there a good place to pull out these constants so that this util can reference them and it won't cause a circular import?

Reply:
> Is there a good place to pull out these constants so that this util can reference

We currently don't have such a module. I think rewriting the constants in this file is ok for now.

Reply:
```python
# TODO: ...
def get_mlodel_and_mlproject_filenames():
    from mlflow.models.model import MLMODEL_FILE_NAME
    from mlflow.projects._project_spec import MLPROJECT_FILE_NAME

    return [MLMODEL_FILE_NAME, MLPROJECT_FILE_NAME]

_TEXT_EXTENSIONS = [
    ...,
    *get_mlodel_and_mlproject_filenames()
]
```

This might work.

Reply:
That still doesn't quite work because _TEXT_EXTENSIONS gets evaluated at import time and it triggers the circular dependency. But tweaking the idea a little, I can put the whole list in a function and that gets around it.

Reply:
Link to tweaked implementation: https://github.com/mlflow/mlflow/pull/8048/files#diff-cc0cc65505c5745295581af3a6ec580c1c734505dc96e748afa4170f09473042R7